jax/_src/sharding.py

```diff
@@ -166,6 +166,11 @@ def devices_indices_map(self, global_shape: Shape) -> Mapping[Device, Index]:
     """
     return common_devices_indices_map(self, global_shape)
 
+  @functools.cached_property
```

Comment:
Hmm, I wonder if there is a different way to spell this. We don't want to expose `client` as a public concept on a public API.

Where are you hitting this? Can you show me the exact repro and the stack trace?